### PR TITLE
Docker Deployment

### PR DESCRIPTION
**Pull request by bodo-hugo-barwich**
This development would enable the Web Application to run in a Docker Container with a SQLite Backend.
Though It has the limitations:

- The `hypnotoad` configuration from `perldoc-browser.conf` has no effect. (The configured port is not used but always port 3000)
- The activity logs are not seen in the container log, only warnings are logged.

For the installation please refer to `etc/docker/README`

Any suggestions or issues are welcomed.

### CODE REVIEW

**Comment by bodo-hugo-barwich** [comment]
#26 This is the first phase to enable Docker Deployment with SQLite Backend.

File: .gitignore [hunk 1]
```diff
@@ -6,3 +6,17 @@ html/*
 perldoc-browser.conf
 perldoc-browser.sqlite*
 hypnotoad.pid
+
+#Project Files
+.gitignore
```
Comment (Grinnz) [inline .gitignore:11]: I don't think this is right, .gitignore is maintained in the repository so should not be ignored by git.

**Comment by Grinnz** [comment]
Why is /etc/docker/README copied in three formats?

**Comment by bodo-hugo-barwich** [comment]
Thank you for reviewing my work.
About the `.gitignore` file I had to include a file from my working environment to make the work easier and I initially did not want to change it.
Later I found that the Container would store its runtime generated files in this directory to and those are not part of the development generally.
If you want to go the Container Update Strategy as I set it up it will generate those files in the working directory. This is convenient for frequent changes in the working directory.
For a production environment I would rather suggest to copy the directory into the container for improved security.

**Comment by bodo-hugo-barwich** [comment]
I included 3 versions of the rendered `README.pod` mostly for online readablity convenience.
I personally found that _GitHub_ renders the `README.md` better than the `.pod` file.
The plain text file `README` is convenient when you clone the project and open it in the local console environment with `vim` as I often do it. Thinking about it you could also read the `README.pod` file with the `perldoc` command with `perldoc README.pod` which would also result into somewhat the same outlay.
`README.md` and `README` are just the rendered results from `README.pod` and I can remove them if you prefer.

**Comment by Grinnz** [comment]
Yes, I would prefer only one file so it can be easily maintained. pod format seems good.

**Comment by bodo-hugo-barwich** [comment]
I agree about the maintainability and I removed the copies of the README file

**Comment by Grinnz** [comment]
Thank you! If you have any further developments please feel free to send additional PRs.

**Pull request merged**